internal/controller/postgrescluster/metrics_setup.sql

```diff
@@ -220,3 +220,150 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql;
 
+/*
+ * Tables and functions for monitoring changes to pg_hba_file_rules system catalogs.
+ * Tables allow recording of existing settings so they can be referred back to to see what changed
+ * If checksum function returns 0, then NO settings have changed
+ * If checksum function returns 1, then something has changed since last known valid state
+ * For replicas, logging past settings is not possible to compare what may have changed
+ * For replicas, by default, it is expected that its settings will match the primary
+ * For replicas, if the pg_hba.conf are necessarily different from the primary, a known good hash of that replica's
+    settings can be sent as an argument to the relevant checksum function. Views are provided to easily obtain the hash values used by this monitoring tool.
```

Comment:
I find this comment difficult to follow.

Reply:
OK, I massively rewrote the comments -- and also some of the function. Once I started writing the comments, it was hard to keep myself from streamlining some of the function.

Reply:
Much appreciated!